apps/api/src/leaderBoard/leaderBoard.service.ts

```diff
@@ -0,0 +1,24 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { TransformUserService } from 'src/TransformUser/TransformUser.service';
+import { Repository } from 'typeorm';
+import { User, Userfront } from 'types';
+
+@Injectable()
+export class LeaderBoardService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    private readonly transfornService: TransformUserService,
+  ) {}
+
+  async getLeaderBoard(): Promise<Userfront[]> {
+    const users = await this.userRepository.find();
```

Comment:
`const users = await this.userRepository.find({
    order: {
        level: "DESC"
    },
})`
is much more efficient. You can then map result :
return users.map((user) => await this.transformService.transform(user))

Reply:
done